xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/ModelBridge.java

```diff
@@ -64,6 +64,17 @@
      */
     boolean delete(DocumentReference documentReference);
 
+    /**
+     * Delete or move to the recycle bin the specified document.
+     * @param documentReference the reference of the document to delete
+     * @param toRecycleBin If {@code true}, the document is moved to the recycle bin (if the recycle bin is not
+     *                    activated, the document is removed permanently nevertheless). If {@code false},
+     *                    the document is removed permanently
+     * @return {@code true} if the document was deleted successfully, {@code false} if the delete failed
+     * @since 12.8RC1
+     */
+    boolean delete(DocumentReference documentReference, boolean toRecycleBin);
```

Comment:
Configuration is about skipping RecycleBin, old core API are about skipping recycle bin, IMO you should always use the same skipRecycleBin parameter to be consistent.

Reply:
Interface (https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-0665e5b7ff47f7f52c23f66192734100R76) and implementation updated (https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-29fdf5e5e3805db76947ad7e90902ac7R187)